### ISSUE TITLE
Due to an open issue with wsl, testrunner is not reachable in LAN

### ISSUE THREAD

**Issue by mlasak** [cta-wave/dpctf-deploy#83]
https://github.com/microsoft/WSL/issues/4983

open since 2020. only workaround for now is to start test runner without ssl and use a simple proxy on windows host machine to terminate ssl there.

example:

```
const fs = require('fs')
const path = require('path')
const hp = require('http-proxy')

const certFolder = 'certs'

hp.createServer({
    target: {
        host: '172.31.146.167',
        port: 8000
    },
    ssl: {
        key: fs.readFileSync(path.join(certFolder, 'key.pem'), 'utf8'),
        cert: fs.readFileSync(path.join(certFolder, 'cert.pem'), 'utf8'),
        ca: fs.readFileSync(path.join(certFolder, 'ca.pem'), 'utf8')
    }
}).listen(8443)


hp.createServer({
    target: {
        host: '172.31.146.167',
        port: 8000
    },
}).listen(8000)

```

whereas the "172.31.146.167" address should be replaced with the one you have on your setup:

outside wsl call:
```
 wsl hostname -I
```

**Comment by louaybassbouss** [comment]
@mlasak  it is possible to replace Node.js proxy with something else like NGINX?

**Comment by mlasak** [comment]
@louaybassbouss yes, any proxy should work. NGINX as well. However, it must be running directly on host system.

**Comment by jpiesing** [comment]
Don't forget that the documentation needs to be updated somewhere to describe this.

**Comment by mlasak** [comment]
@jpiesing a proxy is imo only a hack/workaround and should only be used in case of desperation. As well, it turns out, that it really depends on your local network setup and security measures like (firewalls, DNS rebind protection, etc). The question is how far a documentation should go and what level of knowledge can be expected from person setting up the test runner?

I have a running setup on 
* win10 
* using wsl2
* without docker for desktop (instead docker installed inside wsl2)
* with free DNS name and SSL certificates 

And yes, we will share/update documentation on this

**Comment by jpiesing** [comment]
> @jpiesing a proxy is imo only a hack/workaround and should only be used in case of desperation. As well, it turns out, that it really depends on your local network setup and security measures (like firewalls, DNS rebind protection, etc). The question is how far a documentation should go and what level of knowledge can be expected from person setting up the test runner?
> 
> I have a running setup on
> 
> * win10
> * using wsl2
> * without docker for desktop (instead docker installed inside wsl2)
> * with free DNS name and SSL certificates
> 
> And yes, we will share/update documentation on this

@mlasak IMHO we should provide limited documentation on using Windows + WSL2 without Docker Desktop. 
We should just list things to watch out for and not go into proper depth on how to solve them at the same level as the main supported configurations of Linux and Windows+Docker Desktop.
We have a (old) document on Windows + WSL2 - https://github.com/cta-wave/dpctf-deploy/blob/master/WINDOWS_WSL.md. I think it's enough to review & update that document and perhaps add a few comments about new problems and the type of solution.

Does this sound reasonable?

**Comment by gitwjr** [comment]
@mlasak @FritzHeiden @jpiesing 
It is not clear if this issue can be closed.
### Looking at the current ReadMe documentation, the following is stated: 

Run wsl hostname -I and copy the (first) IP address e.g. 172.24.202.133 (lets call this <wsl_ip>)
Run
netsh.exe interface portproxy add v4tov4 listenport=8000 listenaddress=0.0.0.0 connectport=8000 connectaddress=<wsl_ip>
netsh.exe interface portproxy add v4tov4 listenport=8443 listenaddress=0.0.0.0 connectport=8443 connectaddress=<wsl_ip>

### Not included from @mlasak comment:
 netsh.exe interface portproxy add v6tov4 listenport=8000 listenaddress=0.0.0.0 connectport=8000 connectaddress=(wsl hostname -I)
netsh.exe interface portproxy add v6tov4 listenport=8443 listenaddress=0.0.0.0 connectport=8443 connectaddress=(wsl hostname -I)

docker-compose.yml: network_mode="bridge" NOT "host". 
There is no reference to any additional sources on WSL networking.

**Comment by FritzHeiden** [comment]
The linked issue from the original post is now resolved, though I didn't test if this eliminates requiring to execute the provided commands.
Furthermore, the excluded commands from @mlasak bind ports for IPv6 which the test runner doesn't use. AFAIK the two commands in the readme resolve the descibed issue.

IMO this can be closed.
